pkg/providers/vsphere/template.go

```diff
@@ -18,29 +18,18 @@ import (
 )
 
 func NewVsphereTemplateBuilder(
-	datacenterSpec *anywherev1.VSphereDatacenterConfigSpec,
-	controlPlaneMachineSpec, etcdMachineSpec *anywherev1.VSphereMachineConfigSpec,
-	workerNodeGroupMachineSpecs map[string]anywherev1.VSphereMachineConfigSpec,
 	now types.NowFunc,
 	fromController bool,
 ) *VsphereTemplateBuilder {
 	return &VsphereTemplateBuilder{
-		datacenterSpec:              datacenterSpec,
-		controlPlaneMachineSpec:     controlPlaneMachineSpec,
-		WorkerNodeGroupMachineSpecs: workerNodeGroupMachineSpecs,
-		etcdMachineSpec:             etcdMachineSpec,
```

Comment:
No need to maintain all of these in here, it's in the cluster spec
Also, this complicated a lot the code, having to keep the template in builder in sync with the defaulting/changes done to the cluster spec (this doesn't use pointers)